src/gazebo_camera_manager_plugin.cpp

```diff
@@ -614,34 +614,30 @@ void CameraManagerPlugin::_handle_camera_info(const mavlink_message_t *pMsg, str
 {
     gzdbg << "Send camera info" << endl;
     _send_cmd_ack(pMsg->sysid, pMsg->compid, MAV_CMD_REQUEST_CAMERA_INFORMATION, MAV_RESULT_ACCEPTED, srcaddr);
-    static const char vendor[MAVLINK_MSG_CAMERA_INFORMATION_FIELD_VENDOR_NAME_LEN] = "PX4.io";
-    static const char model[MAVLINK_MSG_CAMERA_INFORMATION_FIELD_MODEL_NAME_LEN] = "Gazebo";
-    static const char uri[MAVLINK_MSG_CAMERA_INFORMATION_FIELD_CAM_DEFINITION_URI_LEN] = {};
-    uint32_t camera_capabilities = CAMERA_CAP_FLAGS_CAPTURE_IMAGE | CAMERA_CAP_FLAGS_CAPTURE_VIDEO |
-            CAMERA_CAP_FLAGS_HAS_MODES | CAMERA_CAP_FLAGS_HAS_BASIC_ZOOM |
-            CAMERA_CAP_FLAGS_HAS_VIDEO_STREAM;
+
+    mavlink_camera_information_t camera_information{};
+    camera_information.time_boot_ms = 0;
+    std::strcpy((char *)camera_information.vendor_name, "PX4.io");
+    std::strcpy((char *)camera_information.model_name, "Gazebo");
+    camera_information.firmware_version = 0x01;
+    camera_information.focal_length = 50.f;
+    camera_information.sensor_size_h = 35.f;
+    camera_information.sensor_size_v = 24.f;
+    camera_information.resolution_h = _width;
+    camera_information.resolution_v = _height;
+    camera_information.lens_id = 0;
+    camera_information.flags = CAMERA_CAP_FLAGS_CAPTURE_IMAGE
+                                | CAMERA_CAP_FLAGS_CAPTURE_VIDEO
+                                | CAMERA_CAP_FLAGS_HAS_MODES
+                                | CAMERA_CAP_FLAGS_HAS_BASIC_ZOOM
+                                | CAMERA_CAP_FLAGS_HAS_VIDEO_STREAM;;
+    camera_information.cam_definition_version = 0;
+    std::strcpy(camera_information.cam_definition_uri, "");
+    camera_information.gimbal_device_id = 0;
```

Comment:
You could skip all these fields you're not actually setting.

Reply:
ok, I left away all the zero assignments in 357396f8c048a8ceadb14cfc6409817b2079d15b